src/main/scala/mesosphere/marathon/core/CoreModule.scala

```diff
@@ -0,0 +1,22 @@
+package mesosphere.marathon.core
+import mesosphere.marathon.core.base.Clock
+import mesosphere.marathon.core.launcher.LauncherModule
+import mesosphere.marathon.core.launchqueue.LaunchQueueModule
+import mesosphere.marathon.core.leadership.LeadershipModule
+import mesosphere.marathon.core.task.bus.TaskBusModule
+import mesosphere.marathon.core.task.tracker.TaskTrackerModule
+
+/**
+  * The exported interface of the [[CoreModuleImpl]].
+  *
+  * This is necessary to allow guice to introduce proxies to break cyclic dependencies
+  * (as long as we have them).
+  */
+trait CoreModule {
```

Comment:
How? As I wrote in the comment, this trait is necessary because of our cyclic dependencies. We should eventually get rid of them but I do not want to extend the scope of this PR to encompass that.